pipeline/snapcraft/snapcraft.go

```diff
@@ -131,11 +134,14 @@ func create(ctx *context.Context, folder, arch string, binaries []context.Binary
 		return err
 	}
 
-	snap := metadata.Name + "_" + metadata.Version + "_" + arch + ".snap"
+	snap := filepath.Join(
+		ctx.Config.Dist,
+		metadata.Name+"_"+metadata.Version+"_"+arch+".snap",
+	)
```

Comment:
This will fail because we are setting cmd.Dir before running the command, so it will try to put the .snap in an unexisting dist dir, in the wrong path.

It will work with this diff:
http://paste.ubuntu.com/25233953/